docs/timezone.md

```diff
@@ -45,6 +47,51 @@ tz1.getTransitions(now).next().done;  // => true
 tz2.getTransitions(now).next().done;  // => false
 ```
 
+## Temporal.TimeZone.from(thing: string | object) : Temporal.TimeZone
+
+**Parameters:**
+- `thing` (string or object): A `Temporal.TimeZone` object or a string from which to create a `Temporal.TimeZone`.
+
+**Returns:** a new `Temporal.TimeZone` object.
+
+This static method creates a new time zone from another value.
+If the value is a string, it can be:
+- a string that is accepted by `new Temporal.TimeZone()`;
+- a string in the ISO 8601 format including a time zone part, optionally extended with an IANA time zone name in square brackets appended to it;
+- or, a string with _only_ the time zone part of the ISO 8601 format, again extended with optional IANA name in brackets.
+Or, if the value is an object, it can be another `Temporal.TimeZone` object, which is returned directly.
+
+This function is often more convenient to use than `new Temporal.TimeZone()` because it handles a wider range of input.
+
+Usage examples:
+```javascript
+// IANA time zone names and UTC offsets
+tz = Temporal.TimeZone.from('UTC');
+tz = Temporal.TimeZone.from('Africa/Cairo');
+tz = Temporal.TimeZone.from('america/VANCOUVER');
+tz = Temporal.TimeZone.from('Asia/Katmandu');  // alias of Asia/Kathmandu
+tz = Temporal.TimeZone.from('-04:00');
+tz = Temporal.TimeZone.from('+645');
+
+// ISO 8601 string with time zone part
+tz = Temporal.TimeZone.from('2020-01-14T00:31:00.065858086Z');
+tz = Temporal.TimeZone.from('2020-01-13T16:31:00.065858086-08:00');
+tz = Temporal.TimeZone.from('2020-01-13T16:31:00.065858086-08:00[America/Vancouver]');
+
+// Only time zone part of ISO 8601 string
+tz = Temporal.TimeZone.from('Z');
+tz = Temporal.TimeZone.from('-08:00[America/Vancouver]');
+// If the offset and IANA time zone name disagree, the name takes precedence
+Temporal.TimeZone.from('-05:00[UTC]').name;  // => UTC
```

Comment:
What?!?

Reply:
Well, we have to do _something_ if they disagree, and this is what the polyfill currently does :stuck_out_tongue_closed_eyes: 

I agree that it seems off, though. Other options would be:
- consider the string invalid if they disagree (although that opens a can of worms because an IANA zone can have multiple offsets throughout the year and throughout history, so what does disagreement mean? We should _definitely_ not allow a bare time zone part if that's the case, since agreement would only be well-defined for a specific date and time)
- change our extension to ISO 8601 so that it's not possible to give both an offset and a name

Reply:
> * We should _definitely_ not allow a bare time zone part if that's the case, since agreement would only be well-defined for a specific date and time)

What do you mean? `Temporal.TimeZone.from('-05:00')` should return the time zone that is always 5 hours behind UTC, `Temporal.TimeZone.from('UTC')` should return the time zone that is always UTC, and `Temporal.TimeZone.from('America/New_York')` should return the US Eastern Time Zone (whose relationship to UTC is variable).

Reply:
> > * We should _definitely_ not allow a bare time zone part if that's the case, since agreement would only be well-defined for a specific date and time)
> 
> What do you mean? `Temporal.TimeZone.from('-05:00')` should return the time zone that is always 5 hours behind UTC, `Temporal.TimeZone.from('UTC')` should return the time zone that is always UTC, and `Temporal.TimeZone.from('America/New_York')` should return the US Eastern Time Zone (whose relationship to UTC is variable).

Sorry, I should have clarified that better; my thinking is that we have two somewhat overlapping possibilities for designating a time zone, which I'm calling
- "time zone identifier" (offset or IANA name), and
- "bare time zone part of ISO 8601 string" (whatever we'd allow as an ISO 8601 string, but with the date and time parts chopped off).
Strings like `UTC` and `America/Vancouver` are only valid "time zone identifiers". Strings like `Z` and `-08:00[America/Vancouver]` are only valid "bare time zone parts". Strings like `-08:00` are valid as both.

My opinion is that because a "bare time zone part" can actually contradict itself (`-05:00[UTC]`) that's another argument for not allowing them when constructing a time zone object. I suspect we agree about this.

> > change our extension to ISO 8601 so that it's not possible to give both an offset and a name
> 
> It's only a problem when constructing a time zone object—in all other cases, time zone names are only valid after UTC offsets, which theirselves are only valid after datetimes (and therefore provide access to sufficient information for detecting disagreement and rejecting the input when found).

Agreed.